src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -214,63 +216,68 @@ void AppHost::_HandleCommandlineArgs(const Remoting::WindowRequestedArgs& window
         _revokers.peasantDisplayWindowIdRequested = _peasant.DisplayWindowIdRequested(winrt::auto_revoke, { this, &AppHost::_DisplayWindowId });
         _revokers.peasantQuitRequested = _peasant.QuitRequested(winrt::auto_revoke, { this, &AppHost::_QuitRequested });
 
-        // This is logic that almost seems like it belongs on the WindowEmperor.
-        // It probably does. However, it needs to muck with our own window so
-        // much, that there was no reasonable way of moving this. Moving it also
-        // seemed to reorder bits of init so much that everything broke. So
-        // we'll leave it here.
-        const auto numPeasants = _windowManager.GetNumberOfPeasants();
-        // Don't attempt to session restore if we're just making a window for tear-out
-        if (!startedForContent && numPeasants == 1)
+        _windowLogic.WindowName(_peasant.WindowName());

```

Comment:
You're gonna want to [suppress whitespace](https://github.com/microsoft/terminal/pull/15263/files?w=1)